fieldpath.go

```diff
@@ -0,0 +1,30 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package celvet
+
+import "k8s.io/apimachinery/pkg/util/validation/field"
+
+const (
+	// NoVersion is the constant that should be passed to linter functions if
+	// the original schema has only one version (the paths output will be
+	// slightly different when this is done).
+	NoVersion = -1
+)
+
+func getRootPath(version int) *field.Path {
```

Comment:
suggestion: *int allows nil to be used to represent an absent version, eliminating the need for the const at the cost of having to construct a pointer for non-nil cases.

Reply:
That's simpler, yeah - fixed!